### PR TITLE
Update distributions task to build embedded Tomcat tar.gz archives only

### PR DESCRIPTION
**Pull request by labkey-adam**
#### Rationale
We only build embedded Tomcat distributions now. Clear out unused properties and options.

#### Related Pull Requests
- https://github.com/LabKey/distributions/pull/473

### CODE REVIEW

**Comment by labkey-adam** [comment]
@labkey-susanh everything builds just fine with all our current `build.gradle` files, but I could use help with one thing: getting `subDirName` optional behavior to work right. If this property is removed from `build.gradle`, I see the following error:
```
FAILURE: Build failed with an exception.

* What went wrong:
A problem was found with the configuration of task ':distributions:starter:distribution' (type 'ModuleDistribution').
  - In plugin 'org.labkey.build.base' type 'org.labkey.gradle.task.ModuleDistribution' property 'artifactId' doesn't have a configured value.

    Reason: This property isn't marked as optional and no value has been configured.

    Possible solutions:
      1. Assign a value to 'artifactId'.
      2. Mark property 'artifactId' as optional.
```

File: src/main/groovy/org/labkey/gradle/task/ModuleDistribution.groovy [hunk 1]
```diff
@@ -28,30 +28,25 @@ import org.labkey.gradle.plugin.extension.DistributionExtension
...
-    @Optional @Input
-    Boolean makeDistribution = true // set to false for just an archive of modules
+
     @Optional @Input
     String extraFileIdentifier = ""
     @Optional @Input
     String versionPrefix = null
     @Optional @Input
-    String subDirName
+    String subDirName = null
```
Comment (labkey-susanh) [inline src/main/groovy/org/labkey/gradle/task/ModuleDistribution.groovy]:
```suggestion
    @Input
    final abstract Property<String> subDirName = project.objects.property(String).convention(project.name)
```

File: src/main/groovy/org/labkey/gradle/task/ModuleDistribution.groovy [hunk 2]
```diff
@@ -80,45 +75,21 @@ class ModuleDistribution extends DefaultTask
     File getDistributionDir()
     {
         if (distributionDir == null) {
-            var subDir = StringUtils.trimToNull(subDirName)
-            if (subDir == null)
-                subDir = project.name
-            distributionDir = project.file("${distExtension.dir}/${subDir}")
+            distributionDir = project.file("${distExtension.dir}/" + getSubDirName())
```
Comment (labkey-susanh) [inline src/main/groovy/org/labkey/gradle/task/ModuleDistribution.groovy]:
```suggestion
            distributionDir = project.file("${distExtension.dir}/" + subDirName.get())
```

File: src/main/groovy/org/labkey/gradle/task/ModuleDistribution.groovy [hunk 3]
```diff
@@ -185,47 +154,13 @@ class ModuleDistribution extends DefaultTask
...
-    {
-        if (shouldBuildStandaloneTarGz())
-            tarArchives()
-        if (shouldBuildEmbeddedArchive(DistributionExtension.TAR_ARCHIVE_EXTENSION))
-            embeddedTomcatTarArchive()
-        if (includeZipArchive)
-            zipArchives()
-        if (shouldBuildEmbeddedArchive(DistributionExtension.ZIP_ARCHIVE_EXTENSION))
-            embeddedTomcatZipArchive()
+        embeddedTomcatTarArchive()
```
Comment (labkey-susanh) [inline src/main/groovy/org/labkey/gradle/task/ModuleDistribution.groovy]: Seems that this method could just be inlined.

File: src/main/groovy/org/labkey/gradle/task/ModuleDistribution.groovy [hunk 4]
```diff
@@ -185,47 +154,13 @@ class ModuleDistribution extends DefaultTask
...
     }
 
     @Input
     String getArtifactId()
     {
-        return subDirName
+        return getSubDirName()
```
Comment (labkey-susanh) [inline src/main/groovy/org/labkey/gradle/task/ModuleDistribution.groovy]:
```suggestion
        return subDirName.get()
```

**Review by labkey-susanh — COMMENTED** [review]
I think using a Property with a convention to provide the default value may get you what you need for the subDirName.

File: src/main/groovy/org/labkey/gradle/task/ModuleDistribution.groovy [hunk 5]
```diff
@@ -247,26 +182,11 @@ class ModuleDistribution extends DefaultTask
...
-    }
-
-    private String getEmbeddedZipArchivePath()
-    {
-        return "${getDistributionDir()}/${getArchiveName()}${DistributionExtension.EMBEDDED_SUFFIX}.${DistributionExtension.ZIP_ARCHIVE_EXTENSION}"
-    }
-
     private String getWarArchivePath()
     {
         return "${getDistributionDir()}/${getArchiveName()}.war"
```
Comment (labkey-tchad) [inline src/main/groovy/org/labkey/gradle/task/ModuleDistribution.groovy]: Maybe remove `getWarArchivePath`. I think we abandoned that a long time ago.

**Review by labkey-tchad — APPROVED** [review]

**Comment by labkey-adam** [comment]
I think everything is fixed up now. Local testing worked with `subDirName` omitted. @labkey-susanh feel free to re-review if you want.

File: src/main/groovy/org/labkey/gradle/task/ModuleDistribution.groovy [hunk 6]
```diff
@@ -22,36 +22,32 @@ import org.gradle.api.Project
...
     @Optional @Input
     String embeddedArchiveType = null
-    @Optional @Input
-    Boolean makeDistribution = true // set to false for just an archive of modules
+
     @Optional @Input
     String extraFileIdentifier = ""
     @Optional @Input
     String versionPrefix = null
     @Optional @Input
```
Comment (labkey-susanh) [inline src/main/groovy/org/labkey/gradle/task/ModuleDistribution.groovy:45]: I think the `@Optional` can/should be removed here. We need some kind of name.

**Review by labkey-susanh — APPROVED** [review]

File: src/main/groovy/org/labkey/gradle/task/ModuleDistribution.groovy (hunk 6 above)
Comment (labkey-adam) [reply src/main/groovy/org/labkey/gradle/task/ModuleDistribution.groovy:45]: Interesting... this was marked `@Optional` before my changes. What about keeping it optional and defaulting to `project.name` a la `subDirName`?

Comment (labkey-susanh) [reply src/main/groovy/org/labkey/gradle/task/ModuleDistribution.groovy:45]: I thought I was pointing to the `@Optional` for `subDirName` with this comment.   Did I misread?

Comment (labkey-adam) [reply src/main/groovy/org/labkey/gradle/task/ModuleDistribution.groovy:45]: Sorry, I was unclear which `@Optional` was being referenced. As for `subDirName`, the code was clearly attempting to make it optional; it was annotated as such and defaulting to `project.name.` Only problem was `getArtifactId()` hadn't gotten the memo. It now correctly defaults to `project.name`, which was my goal and what we discussed over email. But maybe I'm misunderstanding your suggestion or what `@Optional` means...

Comment (labkey-susanh) [reply src/main/groovy/org/labkey/gradle/task/ModuleDistribution.groovy:45]: Oh, my bad. I think we do want to leave it so Gradle/Groovy knows the task doesn't require a value to be provided.

**Pull request merged**